### PR TITLE
Enhance advance command 

### LINKED ISSUE #38: Enhance AdvanceCommand to include date of interview
This applies only for advancing applicants from `Applied` to `Shortlisted`.

### PR DESCRIPTION
**Pull request by FredericChow00**
## What

Enhanes `AdvanceCommand` to filter out when it is necessary for user to input an interview date and when it is not necessary to do so.

## Why
InterviewDateTime input is only necessary by users when they are advancing a user from `Applied` to `Shortlisted`, and not needed for `Shortlisted` to `Accepted`.

This allows the application to inform and alert the user when he/she accidentally inputs an interview date and time when he/she is not supposed to, and vice-versa.

Fix #38

### CODE REVIEW

File: src/main/java/seedu/address/logic/commands/AdvanceCommand.java [hunk 1]
```diff
@@ -51,22 +58,61 @@ public CommandResult execute(Model model) throws CommandException {
...
-            return new CommandResult("Successfully advanced " + personToAdvance.getName().fullName);
+
+        /* this if-statement will check whether the applicant can be advanced and also whether
+        the interview datetime input is valid.
+        If any exception occurs along the way, then CommandException will be thrown */
+        if (canAdvanceApplicant(model, personToAdvance) &&
+                isValidInterviewDateInput(personToAdvance, this.interviewDateTime)) {
+            advanceApplicant(model, personToAdvance, this.interviewDateTime);
+        }
+        return new CommandResult("Successfully advanced " + personToAdvance.getName().fullName);
```
Comment (Merrickneo) [inline src/main/java/seedu/address/logic/commands/AdvanceCommand.java:71]: This violates the Law of Demeter. I believe the code should still be able to run without accessing the fullName attribute

File: src/main/java/seedu/address/model/person/InterviewDateTime.java [hunk 2]
```diff
@@ -8,10 +8,10 @@
 
 /**
  * Represents a Person's interview date and time in the address book.
- * Guarantees: interview's date and time are valid
  */
 public class InterviewDateTime {
-    private static LocalDateTime dateTime;
```
Comment (Merrickneo) [inline src/main/java/seedu/address/model/person/InterviewDateTime.java:14]: Nice spot!

File: src/main/java/seedu/address/model/person/Person.java [hunk 3]
```diff
@@ -102,28 +132,15 @@ public boolean isSamePerson(Person otherPerson) {
...
-        case APPLIED:
-            this.status = Status.SHORTLISTED;
-            return true;
-        case SHORTLISTED:
-            this.status = Status.ACCEPTED;
-            return true;
-        default:
-            return false;
-        }
+    public boolean canAdvance() {
```
Comment (Merrickneo) [inline src/main/java/seedu/address/model/person/Person.java:135]: Nice use of ternary operators to help with code compactness.

**Review by Merrickneo — COMMENTED** [review]
Nice job! Looks like your implementation allows users to omit the datetime field if it is not necessary for the `advance` command.

Seems like some test cases are failing so do take a look at those!

File: src/main/java/seedu/address/logic/commands/AdvanceCommand.java (hunk 1 above)
Comment (FredericChow00) [reply src/main/java/seedu/address/logic/commands/AdvanceCommand.java]: Oops yeah my bad! Will change it

**Pull request merged**